examples/shors_code.py

```diff
@@ -0,0 +1,118 @@
+(0, 8): ───────────────────X───────X───@───────────────────M───
+"""
+
+import random
+
+import cirq
+
+
+class OneQubitShorsCode:
+
+    def __init__(self):
+        self.num_physical_qubits = 9
+        self.physical_qubits = cirq.LineQubit.range(self.num_physical_qubits)
+
+    def encode(self):
+        yield cirq.ops.Moment(
+            [cirq.CNOT(self.physical_qubits[0], self.physical_qubits[3])])
+        yield cirq.ops.Moment(
+            [cirq.CNOT(self.physical_qubits[0], self.physical_qubits[6])])
+        yield cirq.ops.Moment([
+            cirq.H(self.physical_qubits[0]),
+            cirq.H(self.physical_qubits[3]),
+            cirq.H(self.physical_qubits[6])
+        ])
+        yield cirq.ops.Moment([
+            cirq.CNOT(self.physical_qubits[0], self.physical_qubits[1]),
+            cirq.CNOT(self.physical_qubits[3], self.physical_qubits[4]),
+            cirq.CNOT(self.physical_qubits[6], self.physical_qubits[7])
+        ])
+        yield cirq.ops.Moment([
+            cirq.CNOT(self.physical_qubits[0], self.physical_qubits[2]),
+            cirq.CNOT(self.physical_qubits[3], self.physical_qubits[5]),
+            cirq.CNOT(self.physical_qubits[6], self.physical_qubits[8])
+        ])
+
+    def apply_gate(self, gate: cirq.Gate, pos: int):
+        if pos > self.num_physical_qubits:
+            raise IndexError
+        else:
+            return gate(self.physical_qubits[pos])
+
+    def correct(self):
+        yield cirq.ops.Moment([
+            cirq.CNOT(self.physical_qubits[0], self.physical_qubits[1]),
+            cirq.CNOT(self.physical_qubits[3], self.physical_qubits[4]),
+            cirq.CNOT(self.physical_qubits[6], self.physical_qubits[7])
+        ])
+        yield cirq.ops.Moment([
+            cirq.CNOT(self.physical_qubits[0], self.physical_qubits[2]),
+            cirq.CNOT(self.physical_qubits[3], self.physical_qubits[5]),
+            cirq.CNOT(self.physical_qubits[6], self.physical_qubits[8])
+        ])
+        yield cirq.ops.Moment([
+            cirq.CCNOT(self.physical_qubits[1], self.physical_qubits[2],
+                       self.physical_qubits[0]),
+            cirq.CCNOT(self.physical_qubits[4], self.physical_qubits[5],
+                       self.physical_qubits[3]),
+            cirq.CCNOT(self.physical_qubits[7], self.physical_qubits[8],
+                       self.physical_qubits[6])
+        ])
+        yield cirq.ops.Moment([
+            cirq.H(self.physical_qubits[0]),
+            cirq.H(self.physical_qubits[3]),
+            cirq.H(self.physical_qubits[6])
+        ])
+        yield cirq.ops.Moment(
+            [cirq.CNOT(self.physical_qubits[0], self.physical_qubits[3])])
+        yield cirq.ops.Moment(
+            [cirq.CNOT(self.physical_qubits[0], self.physical_qubits[6])])
+        yield cirq.ops.Moment([
+            cirq.CCNOT(self.physical_qubits[3], self.physical_qubits[6],
+                       self.physical_qubits[0])
+        ])
+
+    def measure(self):
```

Comment:
I don't think this is correct either. Measuring the stabilizer formalism should keep the system in the codespace.

Reply:
If it's measurement of syndromes, then yes for sure and should be placed before correction. But Shor's code doesn't need that step. The measurement here is just to read the final result. We actually only need to read the first qubit, so no problem if we want to get rid of measurements of other qubits. I measure all the qubits for debugging reasons.

Reply:
Let's skip the measure method, I think it is confusing, as with this encode/decode form the only qubit of interest to be measured would be the first qubit.